platform-dom/src/main/java/com/softicar/platform/dom/elements/popup/compositor/DomPopupHierarchyGraph.java

```diff
@@ -33,53 +34,73 @@ public DomPopupHierarchyGraph() {
 	 *            the parent {@link DomPopup} (never <i>null</i>)
 	 * @param child
 	 *            the child {@link DomPopup} (never <i>null</i>)
+	 * @throws IllegalArgumentException
+	 *             if the given parent and child references are the same
 	 */
 	public void add(DomPopup parent, DomPopup child) {
 
 		Objects.requireNonNull(parent);
 		Objects.requireNonNull(child);
+		if (parent == child) {
+			throw new IllegalArgumentException();
+		}
 		graph.computeIfAbsent(parent, dummy -> new ArrayList<>()).add(child);
 	}
 
 	/**
-	 * Determines the transitive child {@link DomPopup} nodes of the given
+	 * Determines the transitive child {@link DomPopup} elements of the given
 	 * parent {@link DomPopup}.
 	 * <p>
 	 * The returned {@link List} will be ordered according to descending depth
 	 * in the hierarchy graph. That is, deeply-nested children will occur before
 	 * less deeply-nested children.
+	 * <p>
+	 * Returns an empty {@link List} if the given {@link DomPopup} is not a
+	 * parent element in this graph.
 	 *
 	 * @param parent
 	 *            the parent {@link DomPopup} (never <i>null</i>)
-	 * @return all child {@link DomPopup} nodes (never <i>null</i>)
+	 * @return all child {@link DomPopup} elements (never <i>null</i>)
 	 */
 	public List<DomPopup> getAllChildPopups(DomPopup parent) {
 
+		Objects.requireNonNull(parent);
 		var childPopups = new ArrayList<DomPopup>();
-		List<DomPopup> children = graph.get(parent);
-		if (children != null && !children.isEmpty()) {
-			for (DomPopup child: children) {
-				childPopups.addAll(getAllChildPopups(child));
-			}
+		Optional.ofNullable(graph.get(parent)).ifPresent(children -> {
+			children.stream().map(this::getAllChildPopups).forEach(childPopups::addAll);
 			childPopups.addAll(children);
-		}
+		});
 		return childPopups;
 	}
 
 	/**
-	 * Removes all {@link DomPopup} nodes that are not appended from this graph.
+	 * Removes all {@link DomPopup} elements from this
+	 * {@link DomPopupHierarchyGraph}.
+	 */
+	public void removeAll() {
```

Comment:
why  not a standard name?
```suggestion
	public void clear() {
```

Reply:
Well, I kind of standardized that:
![image](https://user-images.githubusercontent.com/15086658/168668826-cdcbf62d-e2b3-4600-9dd2-5bd74c0ce161.png)
...but yea, "clear" is fine as well.